### ISSUE TITLE
Wrong NULL check

### ISSUE THREAD

**Issue by BugBeginner** [openssl/openssl#19154]
According to the documentation of the function `SSL_CIPHER_get_name()`, it claims that `SSL_CIPHER_get_name() returns a pointer to the name of cipher. If the argument is the NULL pointer, a pointer to the constant value "NONE" is returned.`. However, in code `apps/ciphers.c`, the code is(line 226 to line 235):
```
  if (!verbose) {
          for (i = 0; i < sk_SSL_CIPHER_num(sk); i++) {
              const SSL_CIPHER *c = sk_SSL_CIPHER_value(sk, i);
              p = SSL_CIPHER_get_name(c);
              if (p == NULL)
                  break;
              if (i != 0)
                  BIO_printf(bio_out, ":");
              BIO_printf(bio_out, "%s", p);
          }
```
`SSL_CIPHER_get_name` is unable to return NULL, so it is not necessary to check NULL, maybe to check `p==(None)` is necessary

Besides, for the documentation, should be return "(NONE)" rather than "NONE", which is inconsistent with the code(`ssl/ssl_ciph.c` line 1920 to 1925).
```
const char *SSL_CIPHER_get_name(const SSL_CIPHER *c)
{
    if (c != NULL)
        return c->name;
    return "(NONE)";
}
```

**Comment by BugBeginner** [comment]
Yeah, I do not think it is harmful, either. And I do not think here need to check, if here need to check, why do other code using SSL_CIPHER_get_name not check?

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of the 3.4 dev cycle barring further input
